File: C++/StackQueue.cpp
#include<iostream>
#include<stack>

class Solution {
  public:
    void push(int node) {
        stack1.push(node);
    }

    int pop() {
        if(stack1.empty() && stack2.empty()){
            return -1;
        }
        if(stack2.empty()){
            while(!stack1.empty()){
                int s1 =stack1.top();
                stack1.pop();
                stack2.push(s1);
            }
        }
        int s2 =stack2.top();
        stack2.pop();
        return s2;
    }

  private:
    std::stack<int> stack1;
    std::stack<int> stack2;
};

int main(){
    Solution solution;
    solution.push(1);
    solution.push(2);

    std::cout<<"Pop from queue:";
    std::cout<<solution.pop()<<" ";
    std::cout<<solution.pop()<<std::endl;

    return 0;
}

File: python/stackQueue.py
class Solution:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []
    def push(self, node):
        self.stack1.append(node)
    def pop(self):
        if not self.stack1 and not self.stack2:
            return -1        
        if not self.stack2:
            while self.stack1:
                s1 = self.stack1.pop()
                self.stack2.append(s1)
        s2 = self.stack2.pop()
        return s2
solution=Solution()
solution.push(1)
solution.push(2)
print(solution.pop())
print(solution.pop())